src/QuadVelocityController.cpp

```diff
@@ -213,59 +173,63 @@ bool QuadVelocityController::getTransformAfterTime(
-    ROS_ERROR("ros::ok false while waiting for transform");
+    ROS_ERROR("Exception or ros::ok was false");
     return false;
 }
 
 // Gets the velocity from the transformation information
 // Blocks while waiting for a new transform to calculate the velocity with
-bool QuadVelocityController::getVelocities(geometry_msgs::Twist& return_velocities)
+bool QuadVelocityController::waitForNewVelocities(geometry_msgs::Twist& return_velocities)
 {
 
-    // Get current ROS time
-    ros::Time time = ros::Time::now();
-
-    // Check if we already have a velocity at this time
-    if (have_last_velocity_stamped_ && time == last_velocity_stamped_.header.stamp) {
-        return_velocities = last_velocity_stamped_.twist;
-        return true;
-    }
-
-    // We don't have a velocity at this time yet, so make sure we don't fetch
-    // a transform at the same time as the last one
-    while (ros::ok()
-            && ran_once_
-            && time == last_transform_stamped_.header.stamp) {
-        ros::spinOnce();
-    }
-
-    // Calculate the lastest time between transforms that is allowed.
-    ros::Time latest_time_allowed;
-    if (ran_once_)
+    // Make sure that the current time is more than the last transform stamped time
+    while (wait_for_velocities_ran_once_
+            && ros::Time::now() <= last_transform_stamped_.header.stamp
+            && ros::ok())
     {
-        latest_time_allowed = last_transform_stamped_.header.stamp
-                            + ros::Duration(MAX_TRANSFORM_DIFFERENCE_SECONDS);
-    }
-    else
-    {
-        latest_time_allowed = ros::Time(0);
+        ros::spinOnce();
     }
 
-    // Get a brand new transform. Blocks until a transform after the current time
-    // is received.
+    // Get a brand new transform. Blocks until able to transform at the current time.
     geometry_msgs::TransformStamped transformStamped;
-    bool fetched_transform = getTransformAfterTime(time,
-                                                   transformStamped,
-                                                   latest_time_allowed);
+    bool fetched_transform = waitForTransform(transformStamped);
+
     // Check if the call failed
     if (!fetched_transform)
     {
         ROS_ERROR("Failed to fetch new transform in QuadVelocityController");
-        ran_once_ = false;
+        wait_for_velocities_ran_once_ = false;
         return false;
     }
 
+    // Check if the quad velocity controller has not been run once
+    // Update all intermediate values and make a recursive call to waitForNewVelocities
+    // (meaning another transform will be collected) and a valid velocity can be calculated.
+    if(!wait_for_velocities_ran_once_)
+    {
+        wait_for_velocities_ran_once_ = true;
+        last_transform_stamped_ = transformStamped;
+        return waitForNewVelocities(return_velocities);
+    }
+    // If we have ran before make sure the difference between transforms isn't too much
+    else
```

Comment:
Either the entire body of the function after this should be inside this `else` block, or none of it should.  I vote none of it, if you put all of it, then for consistency, we should put everything after line 229 inside another `else` on that `if` condition.